# Predict Student Earnings -- Microsoft Data Science Capstone Challenge

## Information

- [Homepage of Competition](https://datasciencecapstone.org/competitions/2/student-earnings/page/5/)
- [Problem Description](https://datasciencecapstone.org/competitions/2/student-earnings/page/6/)
- [LeaderBoard of Public Score](https://datasciencecapstone.org/competitions/2/student-earnings/leaderboard/)

## General Idea of Solution

1. Minimum data wrangling
2. Single decision tree regressor as benchmark
3. Iteration for further data wrangling tactic, feature engineering and single model selection
4. Stacking models

## Step 1: Minimum Data Wrangling

### 1.1 Gathering / Importing

In [2]:
%matplotlib inline

import pandas as pd
import numpy as np
import seaborn as sns

In [3]:
X_train = pd.read_csv('train_values.csv', header = 0, index_col = 0)
y_train = pd.read_csv('train_labels.csv', header = 0, index_col = 0)
X_test = pd.read_csv('test_values.csv', header = 0, index_col = 0)

### 1.2 Assessing

In [58]:
# Concatenate X_train and X_test
X_conc = pd.concat([X_train, X_test])
X_conc.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26299 entries, 0 to 26298
Columns: 297 entries, academics__program_assoc_agriculture to student__size
dtypes: float64(287), int64(1), object(9)
memory usage: 59.8+ MB


In [59]:
# Check dtypes
X_conc.dtypes.value_counts()

float64    287
object       9
int64        1
dtype: int64

In [60]:
X_conc.columns[X_train.dtypes=='object'].tolist()

['report_year',
 'school__degrees_awarded_highest',
 'school__degrees_awarded_predominant',
 'school__institutional_characteristics_level',
 'school__main_campus',
 'school__online_only',
 'school__ownership',
 'school__region_id',
 'school__state']

In [97]:
# Find out features with missing values
X_conc.isnull().sum().sort_values(ascending=True).head(10) / len(X_train)

school__institutional_characteristics_level    0.000000
school__state                                  0.000000
school__main_campus                            0.000000
school__degrees_awarded_predominant_recoded    0.000000
school__degrees_awarded_predominant            0.000000
school__degrees_awarded_highest                0.000000
report_year                                    0.000000
school__region_id                              0.000000
school__ownership                              0.000000
student__demographics_age_entry                0.004384
dtype: float64

In [ ]:
# Find out the high correlated features
# Iterative find out corr>0.9

#### Identified Quality and Tidiness Issues

##### Quality (Dirty)

- Completeness
    - only 9 of 297 features has no missing value, and 8 of them are categorical features
    - 
- Validity: none
- Accuracy: none
- Consistency: none

##### Tidiness (Messy): None

##### Learning Algorithm Purpose

- 9 categorical features

### 1.3 Cleaning

## Step 2: Benchmark Buildup

## Step 3: Interation of Further Data Wrangling

## Step 4: Interation of Feature Engineering

## Step 5: Interation of Single Model Selection